wget 'http://sources.redhat.com/cgi-bin/cvsweb.cgi/libc/sysdeps/mips/dl-machine.h.diff?r1=1.57&r2=1.58&cvsroot=glibc'

ChangeLog:
    (ELF_MACHINE_BEFORE_RTLD_RELOC): Don't use label at end of compound statement.

Fixes error

rtld.c: In function `_dl_start':
rtld.c:183: error: label at end of compound statement
make[2]: *** [/home/dank/wk/crosstool-0.28-rc32/build/mipsel-unknown-linux-gnu/gcc-3.4.0-glibc-2.2.5/build-glibc/elf/rtld.os] Error 1


===================================================================
RCS file: /cvs/glibc/libc/sysdeps/mips/dl-machine.h,v
retrieving revision 1.56
retrieving revision 1.57
diff -u -r1.56 -r1.57
--- libc/sysdeps/mips/dl-machine.h	2002/01/17 23:21:33	1.56
+++ libc/sysdeps/mips/dl-machine.h	2002/01/29 02:58:00	1.57
@@ -1,5 +1,5 @@
 /* Machine-dependent ELF dynamic relocation inline functions.  MIPS version.
-   Copyright (C) 1996,1997,1998,1999,2000,2001 Free Software Foundation, Inc.
+   Copyright (C) 1996-2001, 2002 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
    Contributed by Kazumoto Kojima <kkojima@info.kanagawa-u.ac.jp>.
 
@@ -132,7 +132,7 @@
   got = (ElfW(Addr) *) D_PTR (map, l_info[DT_PLTGOT]);			\
 									\
   if (__builtin_expect (map->l_addr == 0, 1))				\
-    goto done;								\
+    break;								\
 									\
   /* got[0] is reserved. got[1] is also reserved for the dynamic object	\
      generated by gnu ld. Skip these reserved entries from		\
@@ -169,7 +169,6 @@
       got++;								\
       sym++;								\
     }									\
-done:									\
 } while(0)
 
 
